### PR TITLE
fixing todays crashes

### PR DESCRIPTION
**Pull request by adrianschroeter**
due to inconsitent data or delayed job use of models

### CODE REVIEW

File: src/api/app/jobs/project_create_auto_cleanup_requests.rb [hunk 1]
```diff
@@ -60,8 +60,7 @@ def autoclean_project(prj)
                                        <accept_at>' + @cleanupTime.to_s + '</accept_at>
                                      </request>')
     req.save!
-    notify = req.notify_parameters
-    Event::RequestCreate.create notify
+    Event::RequestCreate.create req.notify_parameters
```
Comment (ChrisBr) [inline src/api/app/jobs/project_create_auto_cleanup_requests.rb]: ``Event::RequestCreate.create(req.notify_parameters)``

File: src/api/app/models/bs_request.rb [hunk 2]
```diff
@@ -755,7 +755,7 @@ def notify_parameters(ret = {})
     ret[:description] = self.description
     ret[:state] = self.state
     ret[:oldstate] = self.state_was if self.state_changed?
-    ret[:who] = User.current.login
+    ret[:who] = self.commenter unless self.commenter.blank?
```
Comment (ChrisBr) [inline src/api/app/models/bs_request.rb]: ``if self.commenter.present?``

File: src/api/app/models/package.rb [hunk 3]
```diff
@@ -737,7 +737,11 @@ def delete_on_backend
       path = source_path
       path << Suse::Backend.build_query_from_hash({ user: User.current.login, comment: commit_opts[:comment], requestid: commit_opts[:request]},
                                                   [:user, :comment, :requestid])
-      Suse::Backend.delete path
+      begin
+        Suse::Backend.delete path
+      rescue ActiveXML::Transport::NotFoundError
+        # ignore this error, backend was out of sync
```
Comment (ChrisBr) [inline src/api/app/models/package.rb]: I'd at least log the exception!

File: src/api/app/models/project.rb [hunk 4]
```diff
@@ -845,7 +845,11 @@ def delete_on_backend
     if CONFIG['global_write_through']
       path = source_path
       path << Suse::Backend.build_query_from_hash({user: User.current.login, comment: @commit_opts[:comment]}, [:user, :comment])
-      Suse::Backend.delete path
+      begin
+        Suse::Backend.delete path
+      rescue ActiveXML::Transport::NotFoundError
+        # ignore this error, backend was out of sync
```
Comment (ChrisBr) [inline src/api/app/models/project.rb]: same here!

**Comment by ChrisBr** [comment]
Otherwise LGTM

File: src/api/app/models/bs_request.rb (hunk 2 above)
Comment (adrianschroeter) [inline src/api/app/models/bs_request.rb]: since it can be transfered from backend I think it is safer to ignore also empty strings

Comment (ChrisBr) [inline src/api/app/models/bs_request.rb]: ``!obj.blank? == obj.present?``
I just prefer the if here!

Comment (adrianschroeter) [inline src/api/app/models/bs_request.rb]: okay, adapted

**Pull request merged**